function_pipe/core/function_pipe.py

```diff
@@ -70,6 +70,38 @@ def binary(lhs: FN, rhs: tp.Any) -> FN:
     return binary
 
 
+def _from_this_module(some_tb: types.TracebackType) -> bool:
+    return some_tb.tb_frame.f_code.co_filename == __file__
+
+
+def _exception_with_cleaned_tb(original_exception: BaseException) -> BaseException:
+    """
+    Return back a new exception, where traceback is pointing to the first frame with code originating outside this module.
+    """
+    tb = original_exception.__traceback__
+    assert tb is not None
+    tb_next = None
+
+    assert _from_this_module(tb)  # Sanity
+
+    while True:
+        tb_next = tb.tb_next
+
+        # If `tb_next` is None, it means everything from the stack came from this module.
+        # Use the original exception
+        if tb_next is None:
+            return original_exception
+
+        # If `tb_next` originates from outside the module, it means we are done looking
+        if not _from_this_module(tb_next):
+            break
+
+        # We are still observing frames from inside the module; keep looking
+        tb = tb_next
+
+    return type(original_exception)(*original_exception.args).with_traceback(tb_next)
```

Comment:
Not sure if this is better, but I try to avoid the all-mighty `type` function where I can...
```suggestion
    return original_exception.__class__(*original_exception.args).with_traceback(tb_next)
```